src/component.d.ts

```diff
@@ -0,0 +1,32 @@
+// Type definitions for egjs-component 2.0.0+
+// Project: https://github.com/naver/egjs-component
+// Definitions by: Naver <https://github.com/naver>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace eg;
+
+export = Component;
+declare class Component {
+  constructor(options?: { [key: string]: any });
+  trigger(eventName: string, customEvent?: { [key: string]: any }): boolean;
+  hasOn(eventName: string): boolean;
+  on(eventName: string, handlerToAttach: (event?: { [key: string]: any }) => any): Component;
```

Comment:
A Event always come in as parameters. So.. All `events` should be changed to` event`.
`event?` -> `event`